### PR TITLE
Limit gmail adapter walkthrough to gmail-specific stuff.

### PR DESCRIPTION
**Pull request by mstemm**
Now that there's a general doc on writing
adapters (https://github.com/juttle/juttle/pull/505), remove all the
general stuff from this walkthrough, keeping just the general stuff.

@dmehra

### CODE REVIEW

**Comment by mstemm** [comment]
ping! I'd like to incorporate this in the juttle-gmail-adapter release I'm doing today.

File: docs/write.md [hunk 1]
```diff
@@ -1,57 +1,12 @@
...
 
-This page describes details of the Gmail Adapter's `write gmail` proc, including how the adapter receives the output of programs, how it interacts with the juttle program to signal that all data has been written, and details on the implementation of the juttle proc.
-
-## Program output == JSON points
-
-An adapter receives arrays of JSON points from a given juttle program via the `write` method. Write should save those points to its associated backend in a timely manner and in a way that does not starve the program for resources.
-
-The Gmail Adapter buffers points in memory until either the program has completed or a configurable batch size has been reached. This means that program output may be split across multiple messages.
-
-In the case of the Gmail Adapter, it constructs emails containing the JSON points and sends those emails using the Gmail API. It does so using asynchronous functions (specifically [Bluebird Promise](http://bluebirdjs.com) chains) that do not block the node.js event loop.
```
Comment (dmehra) [inline docs/write.md]: I'd keep this paragraph, it's gmail specific and useful.

File: docs/adapter_impl_notes.md [hunk 2]
```diff
@@ -1,67 +1,17 @@
...
-
-```Javascript
-module.exports = require('./lib/');
-```
-
-When the adapter is loaded, the CLI/juttle-engin perform a `require` of the module (i.e. the top-level directory containing `index.js`). `lib/index.js` in turn `require()s` `read.js` and `write.js`, which contain the implementation of the read and write classes, respectively.
-
-The main function exported by `lib/index.js` takes a `config` argument containing the configuration object for the adapter, and returns an object with `name`, `read`, and `write` attributes.  The value for the `name` attribute is `gmail`, corresponding to the `read gmail`/`write gmail` proc in juttle programs. The value for `read` is a class inheriting from `AdapterRead`, which performs the read work of the adapter. The value for `write` is an class inheriting from `AdapterWrite`, which performs the write work of the adapter.
-
-Here's a slighly simplified version of the exported function from `lib/index.js`:
```
Comment (dmehra) [inline docs/adapter_impl_notes.md]: How about keeping this (with the code snippet below), it seems useful for understanding. Move it to after line 15?

**Comment by dmehra** [comment]
+1 with two suggestions above. Also, since the remaining docs are now quite short, consider making the read and write sections in adapter_impl_notes.md rather than separate files.

**Pull request merged**